src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { useContext, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../stor/Context';
+import classNames from 'classnames';
+
+type Props = {
+  todo: Todo;
+  isProcessed: boolean;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo, isProcessed }) => {
+  const { completed, title } = todo;
+
+  const { updateTodo, deleteTodo, setIsSubmitting, todoRef } =
+    useContext(TodosContext);
+
+  const [editing, setEditing] = useState(false);
+  const [value, setValue] = useState(title);
+
+  function togelStatement() {
+    const newTodo = { ...todo, completed: !todo.completed };
+
+    updateTodo(newTodo);
+  }
+
+  function editingTodo(event: React.FormEvent) {
+    event.preventDefault();
+
+    const trimTitle = value.trim();
+
+    if (trimTitle === title) {
+      setEditing(false);
+
+      return;
+    }
+
+    if (trimTitle) {
+      updateTodo({ ...todo, title: value.trim() }).then(() =>
+        setEditing(false),
+      );
+    } else {
+      deleteTodo(todo);
+    }
+  }
+
+  function hendlerSubmitting() {
+    setIsSubmitting(true);
+    deleteTodo(todo);
+  }
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames('todo', {
+        completed: completed,
+      })}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          onClick={togelStatement}
+          readOnly
+          checked={completed}
+        />
+      </label>
+
+      {!editing ? (
+        <span
+          data-cy="TodoTitle"
+          className="todo__title"
+          onDoubleClick={() => setEditing(true)}
+        >
+          {title}
+        </span>
+      ) : (
+        <form onSubmit={editingTodo}>
+          <input
+            autoFocus
+            data-cy="TodoTitleField"
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            defaultValue={title}
+            onChange={event => setValue(event.target.value)}
+            onBlur={editingTodo}
+            ref={todoRef}
+            onKeyDown={event => {
+              if (event.key === 'Escape') {
+                setEditing(false);
+              }
+            }}
```

Comment:
Pls move the function inside `onKeyDown` handler to a separate const and use as a callback here